src/util.h

```diff
@@ -531,19 +531,26 @@ void zeek_strerror_r(int zeek_errno, char* buf, size_t buflen);
 /**
  * Escapes bytes in a string that are not valid UTF8 characters with \xYY format. Used
  * by the JSON writer and BIF methods.
- * @param val the input string to be escaped
+ * @param val the character data to be escaped. this is a string_view, but it can hold
+ * null characters in the middle of the data.
+ * @param escape_printable_controls flag for whether printable control characters
+ * (e.g. newlines, tabs, etc) should be escaped in the output. Defaults to false.
+ * @param escape_other_controls flag for whether non-printable control characters
+ * (e.g. nulls, bells, etc) should be escaped in the output.
  * @return the escaped string
  */
-std::string json_escape_utf8(const std::string& val, bool escape_printable_controls = true);
+std::string escape_utf8(std::string_view val, bool escape_printable_controls = false,
+                        bool escape_other_controls = true);
```

Comment:
I'm just skimming and nothing to hold this off, but leaving a comment here anyhow.

IMO we should avoid adding more APIs with trailing `bool` parameters and even more if they have defaults. Or, enforce the awkward `/*escape_printable_controls=*/false` style at call sites.

If you're new to Zeek's code base, these trailing `false, false` or `true` are the small quests that keep you on your toes:


>        case TYPE_FUNC: {
>            writer.String(util::escape_utf8({val->val.string_val.data, static_cast<size_t>(val->val.string_val.length)},
>                                            false, false));

>                 std::string result = util::escape_utf8({s, static_cast<size_t>(e - s)}, true);